lib/ver/Makefile

```diff
@@ -16,7 +16,10 @@ CLEAN_CMDS=$(RM) CBuildInfo.cc
 BUILD_YEAR=$(shell date '+%Y')
 USER_NAME_CMD=id | awk -F')' '{ print $$1 }' | awk -F'(' '{ print $$2 }'
 USER_NAME=$(shell $(USER_NAME_CMD))
-PRODUCT_VERSION=$(shell cat $(CPP_SRC_HOME)/gradle.properties | grep '^elasticsearchVersion' | awk -F= '{ print $$2 }' | xargs echo)
+PRODUCT_VERSION:=$(shell cat $(CPP_SRC_HOME)/gradle.properties | grep '^elasticsearchVersion' | awk -F= '{ print $$2 }' | xargs echo)
+ifneq ($(VERSION_QUALIFIER),)
+PRODUCT_VERSION:=$(PRODUCT_VERSION)-$(VERSION_QUALIFIER)
+endif
 ifeq ($(SNAPSHOT),yes)
```

Comment:
Suggestion: Instead of SNAPSHOT and VERSION_QUALIFIER both could be passed as a single string VERSION_QUALIFIER_SNAPSHOT (or just VERSION_QUALIFIER) then both wouldn't need to be passed in the gradle tasks (make, clean, etc)

Reply:
I think it's better to keep the two variables separate so there's a one-to-one correspondence between the Gradle side and the `make` side.  For example, it will make things easier for the release-manager team when they have to set similar variables for the Docker scripts.

I think there is another way to avoid the duplication, by using `configure` and a closure.  I'll give it a go and push another commit if it works.

Reply:
I couldn't get `configure` and a closure to work, but environment variables can be set as a map as well as individually, so I created a single project-wide map first and then each `Exec` task just has one line to use it.